build_image.py

```diff
@@ -224,8 +257,19 @@ def create_dockerfile_finish(config: OfrakImageConfig) -> str:
             "\\n",
         ]
     )
-    dockerfile_finish_parts.append(f'RUN printf "{develop_makefile}" >> Makefile\n')
-    dockerfile_finish_parts.append("RUN make $INSTALL_TARGET\n\n")
+    dockerfile_finish_parts += [
+        f'RUN printf "{develop_makefile}" >> Makefile\n\n',
+    ]
+    if config.check_python_reqs:
+        dockerfile_finish_parts += [
+            '# We use --network="none" to ensure all dependencies were installed in base.Dockerfile\n',
+            'RUN --network="none" make $INSTALL_TARGET\n',
+            "RUN python3 -m pip check\n\n",
```

Comment:
It's not clear to me that this belongs here.

If we decide to enforce that ofrak packages are installed from source (and not PyPI), then that seems like something that belongs in the Makefile.

Similarly, it's not clear to me that checking pip dependencies should happen as a build step -- this feels more like a test/validation step that happens after the image is built (again, if we decide that we want to do that).

The objective of this PR was to optimize Docker build -- I think that is accomplished without this `check_python_reqs` option. What do you think about removing it from this PR and discussing/tackling that as a separate issue?

Reply:
@whyitfor makes sense for `python3 -m pip check`, but `--network="none"` can only be done in a `Dockerfile`. ``python3 -m pip check` only helps catch mismatching dependencies (so yeah, maybe belongs in CI), but `--network="none"` also catches the case where the Docker build is not fully optimized (that is, dependencies that we'd want to install in `base.Dockerfile` were not installed there). But will take it out, and move to a separate PR.